frame/message-queue/src/tests.rs

```diff
@@ -1032,7 +1035,7 @@ fn footprint_on_swept_works() {
 
 #[test]
 fn execute_overweight_works() {
-	new_test_ext::<Test>().execute_with(|| {
+	build_and_execute::<Test>(|| {
```

Comment:
Looks like there is indeed a bug for overweight messages not being handled correctly.

Reply:
Not sure what is happening here.

AFAIK Permanently overweight messages are ignored and the index moves onto next message to be processed. When all processable messages are processed, the queue is unknit from ReadyRing.

But in a special case where the queue present in ReadyRing has only 1 message and it is permanently overweight, then the Queue gets stuck in ReadyRing and queue never gets unknit.
What should we do?

Reply:
You can just fix the other things first, i will open an MR to change the MQ pallet.  
But its great that you found a bug in it; exactly the reason to have these `try_state` checks.

Reply:
Should be fixed here https://github.com/paritytech/substrate/pull/13528